File: Editorial/19_Remove_Nth_Node_From_End_of_List/C/1_Two_pass_algorithm.c
struct ListNode* removeNthFromEnd(struct ListNode* head, int n) {
    struct ListNode* dummy = (struct ListNode*)malloc(sizeof(struct ListNode));
    dummy->next = head;
    int length = 0;
    struct ListNode* first = head;
    while (first != NULL) {
        length++;
        first = first->next;
    }
    length -= n;
    first = dummy;
    while (length > 0) {
        length--;
        first = first->next;
    }
    first->next = first->next->next;
    return dummy->next;
}

File: Editorial/0019_Remove_Nth_Node_From_End_of_List/Go/1_Two_pass_algorithm.go
func removeNthFromEnd(head *ListNode, n int) *ListNode {
    dummy := &ListNode{}
    dummy.Next = head
    length := 0
    first := head
    for first != nil {
        length++
        first = first.Next
    }
    length -= n
    first = dummy
    for length > 0 {
        length--
        first = first.Next
    }
    first.Next = first.Next.Next
    return dummy.Next
}
